### PR TITLE
remove unnecessary mutex locking in the stream flow controller

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: internal/flowcontrol/base_flow_controller.go [hunk 1]
```diff
@@ -60,10 +60,8 @@ func (c *baseFlowController) sendWindowSize() protocol.ByteCount {
 	return c.sendWindow - c.bytesSent
 }
 
+// needs to be called with locked mutex
```
Comment (lucas-clemente) [inline internal/flowcontrol/base_flow_controller.go:63]: It shouldn't be a public fn then, no?

**Review by lucas-clemente — COMMENTED** [review]
GH is showing some unused failures?

File: internal/flowcontrol/base_flow_controller.go (hunk 1 above)
Comment (marten-seemann) [reply internal/flowcontrol/base_flow_controller.go:63]: That probably applies to all function of the `baseFlowController`.

Comment (marten-seemann) [reply internal/flowcontrol/base_flow_controller.go:63]: Done.

**Pull request merged**